src/Hive.IO/Hive.IO/EnergySystems/CombustionTech.cs

```diff
@@ -24,33 +22,34 @@ public GasBoiler(double investmentCost, double embodiedGhg, double capacity, dou
         }
 
 
-        protected override double[] SetConversionEfficiencyHeating()
-        {
-            throw new System.NotImplementedException();
-        }
-
-
-        //where is gas defined?? for solar, its clear, its from a weather file...
-        // so all Input Carriers should be part of Hive.IO.Environment!
-        // indicate, whether we have biogas, wood pellets, district heating, electricity grid, natural gas, oil, etc
-        public void SetInput(Gas gasInput)
+        public void SetInputOutput(Gas gasInput, double[] heatingGenerated, double[] supplyTemperature)
         {
-            base.InputCarrier = gasInput;
-        }
+            int horizon = heatingGenerated.Length;
+
+            var gasConsumed = new double[horizon];
+            var gasPrice = new double[horizon];
+            var gasEmissionsFactor = new double[horizon];
+
+            for (int t = 0; t < horizon; t++)
+            {
+                gasConsumed[t] = heatingGenerated[t] / this.Efficiency;
+            }
+            if (horizon == Misc.MonthsPerYear)
+            {
+                gasPrice = Misc.GetAverageMonthlyValue(gasInput.EnergyPrice);
+                gasEmissionsFactor = Misc.GetAverageMonthlyValue(gasInput.GhgEmissionsFactor);
+            }
+            else
+            {
+                gasPrice = gasInput.EnergyPrice;
+                gasEmissionsFactor = gasInput.GhgEmissionsFactor;
+            }
+
+            Gas gasConsumedCarrier = new Gas(horizon, gasConsumed, gasPrice, gasEmissionsFactor);
```

Comment:
ReSharper normally invites you to just do `base.InputCarrier = new Gas(...)` - you don't really need that variable anymore, right?